packages/clerk-js/src/core/clerk.ts

```diff
@@ -1098,6 +1100,13 @@ export default class Clerk implements ClerkInterface {
         void this.handleUnauthenticated({ broadcast: false });
       }
     });
+
+    // set cookie on token update
+    eventBus.addEventListener('token:update', event => {
+      const tokenEvent = event as CustomEvent;
+      const token = tokenEvent.detail.token as TokenResource;
```

Comment:
```suggestion
      const token = tokenEvent?.detail?.token as TokenResource;
```

Reply:
If token is not available we should noop

Reply:
`event` and `token` should be available but the content of the token could be empty. I have added types to enforce this.
see: 5d67113ea7e6ec42987234fac948739462da8ee4